sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClientBuilder.java

```diff
@@ -0,0 +1,326 @@
+     * @throws NullPointerException if {@code endpoint} has not been set. This is automatically set when {@link
+     *     #connectionString(String) connectionString} is set. Or, explicitly through {@link #endpoint(String)}.
+     * @throws IllegalStateException If {@link #connectionString(String) connectionString} has not been set.
+     */
+    public ServiceBusManagementAsyncClient buildAsyncClient() {
+        if (endpoint == null) {
+            throw new NullPointerException("'endpoint' cannot be null.");
+        }
+
+        final HttpPipeline httpPipeline = buildPipeline();
+        final ServiceBusManagementClientImpl client = new ServiceBusManagementClientImplBuilder()
+            .pipeline(httpPipeline)
+            .serializer(serializer)
+            .endpoint(endpoint)
+            .apiVersion(API_VERSION)
+            .buildClient();
+
+        return new ServiceBusManagementAsyncClient(client, serializer);
+    }
+
+    /**
+     * Creates a {@link ServiceBusManagementClient} based on options set in the builder. Every time {@code
+     * buildAsyncClient} is invoked, a new instance of the client is created.
+     *
+     * <p>If {@link #pipeline(HttpPipeline) pipeline} is set, then the {@code pipeline} and
+     * {@link #endpoint(String) endpoint} are used to create the {@link ServiceBusManagementClient client}. All other
+     * builder settings are ignored.</p>
+     *
+     * @return A {@link ServiceBusManagementClient} with the options set in the builder.
+     * @throws NullPointerException if {@code endpoint} has not been set. This is automatically set when {@link
+     *     #connectionString(String) connectionString} is set. Or it can be set explicitly through {@link
+     *     #endpoint(String)}.
+     * @throws IllegalStateException If {@link #connectionString(String) connectionString} has not been set.
+     */
+    public ServiceBusManagementClient buildClient() {
+        return new ServiceBusManagementClient(buildAsyncClient());
+    }
+
+    /**
+     * Adds a policy to the set of existing policies that are executed after required policies.
+     *
+     * @param policy The retry policy for service requests.
+     *
+     * @return The updated {@link ServiceBusManagementClientBuilder} object.
+     * @throws NullPointerException If {@code policy} is {@code null}.
+     */
+    public ServiceBusManagementClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy);
+        policies.add(policy);
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint for the Service Bus namespace.
+     *
+     * @param endpoint The URL of the Service Bus namespace.
+     *
+     * @return The updated {@link ServiceBusManagementClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public ServiceBusManagementClientBuilder endpoint(String endpoint) {
+        try {
```

Comment:
Should we check for `endpoint` could be null and throw NPE ?

Reply:
It checks on buildClient